internal/core/user/identity/provider.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package identity
+
+import (
+	"reflect"
+
+	"github.com/jinzhu/gorm"
+
+	"github.com/erda-project/erda-infra/base/logs"
+	"github.com/erda-project/erda-infra/base/servicehub"
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/pkg/discover"
+	"github.com/erda-project/erda/pkg/http/httpclient"
+)
+
+type config struct {
+	OryEnabled           bool   `default:"false" file:"ORY_ENABLED"`
+	UCClientID           string `default:"dice" file:"UC_CLIENT_ID"`
+	UCClientSecret       string `default:"secret" file:"UC_CLIENT_SECRET"`
+	OryKratosPrivateAddr string `default:"kratos-admin" file:"ORY_KRATOS_ADMIN_ADDR"`
+}
+
+type Interface interface {
+	FindUsers(ids []string) ([]User, error)
+	FindUsersByKey(key string) ([]User, error)
+	GetUser(userID string) (*User, error)
+	GetUsers(IDs []string, needDesensitize bool) (map[string]apistructs.UserInfo, error)
+	UserMigration(req OryKratosCreateIdentitiyRequest) (string, error)
```

Comment:
move these last two inteface to kratos privately.

Reply:
done